text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,370 @@
+the best approach.
+
+![Development Strategies to Support](./img/0240-dev-strategies.png)
+
+#### Decentralized Approach
+
+Library designers using this schema-first telemetry schema approach add a phase
+in their CI/CD pipeline for the registration of the resolved telemetry schema
+associated with their library for a given version. This registration is done
+towards a registry maintained by OpenTelemetry.
+
+Application/service owners configure their CI/CD pipeline to automatically
+import the resolved telemetry schemas of dependencies (if they exist) in order
+to apply locally the telemetry schema policies or any other build step that
+applies to/feeds off the resolved telemetry schemas of the dependencies.
+
+![Decentralized Approach](./img/0240-decentralized-approach.png)
+
+This approach is considered decentralized as it does not impose any direct
+collaboration between parties while ensuring consistency between versions of
+dependencies and their associated telemetry schemas.
+
+#### Centralized Approach
+
+In this approach, the library designers have no interaction with the external
+world. They define a telemetry schema for their library, version the resolved
+schema in their repository, and release their library as usual.
+
+On the application and service owners' side, they must either make a local copy
+of the resolved telemetry schemas corresponding to their dependencies (similar
+conceptually to the model followed by protobuf and gRPC) or link to the resolved
+telemetry schema for a specific version. In both cases, the synchronization
+between the versions of the library and the resolved telemetry schema is the
+responsibility of the application owner.
+
+![Centralized Approach](./img/0240-centralized-approach.png)
+
+This approach is considered centralized as the application owner has the
+responsibility of maintaining, in a centralized manner locally, all the
+telemetry schemas of the dependencies.
+
+A specific OTEP dedicated to the resolved telemetry schema will be responsible
+for defining the format and structure of the resolved telemetry schema, as well
+as the best approach to collect the resolved telemetry schemas of the different
+dependencies.
+
+## Roadmap
+
+To facilitate the review process and progressively deliver value to the
+OpenTelemetry community, a series of OTEPs and tools are proposed.
+
+### OTEPs
+
+* Telemetry Schema versions (incremental steps):
+  * V1.1.1: Add a section to import one or several Semantic Convention
+  Registries. Once implemented, the official Semantic Convention Registry will
+  be referenced from the official OpenTelemetry Root Telemetry Schema.
+  * V1.2: Add references to parent schemas and define a schema section to
+  describe application/library signals. In this version, only metrics, logs, and
+  spans signals are described. This is the first version of the telemetry schema
+  that can be used by the Client SDK generator. In this version only one schema
```

Comment:
>This is the first version of the telemetry schema
>  that can be used by the Client SDK generator.

Why can't V1.1.1 be used by the generator? I think it will contain enough information to generate code for Otel-wide conventions.

Reply:
See https://github.com/open-telemetry/oteps/pull/243/commits/b1954142ff280067f16185d3f7c89e7985902bc3